tests/legacy-cli/e2e/tests/update/update-10.ts

```diff
@@ -0,0 +1,62 @@
+import { SemVer } from 'semver';
+import { createProjectFromAsset } from '../../utils/assets';
+import { expectFileMatchToExist, readFile } from '../../utils/fs';
+import { setRegistry } from '../../utils/packages';
+import { ng, noSilentNg } from '../../utils/process';
+import { isPrereleaseCli, useCIChrome, useCIDefaults, NgCLIVersion } from '../../utils/project';
+
+export default async function () {
+  try {
+    // We need to use the public registry because in the local NPM server we don't have
+    // older versions @angular/cli packages which would cause `npm install` during `ng update` to fail.
+    await setRegistry(false);
+    await createProjectFromAsset('10.0-project', true);
+
+    // CLI proiject version
+    const { version: cliVersion } = JSON.parse(
+      await readFile('./node_modules/@angular/cli/package.json'),
+    );
+    const cliMajorProjectVersion = new SemVer(cliVersion).major;
+
+    // CLI current version.
+    const cliMajorVersion = NgCLIVersion.major;
+
+    for (let version = cliMajorProjectVersion + 1; version < cliMajorVersion; version++) {
+      // Run all the migrations until the current build major version - 1.
+      // Example: when the project is using CLI version 10 and the build CLI version is 14.
+      // We will run the following migrations:
+      // - 10 -> 11
+      // - 11 -> 12
+      // - 12 -> 13
+      const { stdout } = await ng('update', `@angular/cli@${version}`, `@angular/core@${version}`);
+      if (!stdout.includes("Executing migrations of package '@angular/cli'")) {
```

Comment:
Should we check the exit code as well or is that implicit in the `ng` command? Is there a scenario where `ng update` exits with code 0 but doesn't run the migration?

Reply:
Only when there aren’t any migrations, which is unlikely.

Reply:
Ok, that's probably fine to leave it, though we should probably include a comment that we're verifying at least one migration was executed.